distributed/scheduler.py

```diff
@@ -4550,6 +4567,10 @@ async def add_client(
         We listen to all future messages from this Comm.
         """
         assert client is not None
+        assert client not in self.clients, f"Client {client!r} already exists"
```

Comment:
The changes from fb8f1c900874fd4dd39ac7b433d8678d868b86c5 are a driveby (@graingert and I discussed how the current logic is messed up in potentially allowing the wrong client batchedsend to be deleted and closed). I'm happy to pull them out to a separate PR.